frontend/discourse/tests/acceptance/transformers/composer-force-editor-mode-test.js

```diff
@@ -0,0 +1,51 @@
+import { visit } from "@ember/test-helpers";
+import { test } from "qunit";
+import { withPluginApi } from "discourse/lib/plugin-api";
+import { acceptance } from "discourse/tests/helpers/qunit-helpers";
+
+acceptance("composer-force-editor-mode transformer", function (needs) {
+  needs.user();
+  needs.settings({
+    rich_editor: true,
+    allow_uncategorized_topics: true,
+  });
+
+  test("forces markdown mode and hides toggle", async function (assert) {
+    withPluginApi((api) => {
+      api.registerValueTransformer(
+        "composer-force-editor-mode",
+        ({ context }) => {
+          if (context.model?.action === "createTopic") {
+            return "markdown";
```

Comment:
I mentioned on chat, the actual expected values for `forceEditorMode` is confusing, this should be more clear. Some places are using the values from this `USER_OPTION_COMPOSITION_MODES.rich` "enum", the markdown equivalent is `0`:

https://github.com/discourse/discourse/blob/8a249dc95955cb4a9b43f199f24ea5949e29102f/plugins/automation/admin/assets/javascripts/admin/components/fields/da-post-field.gjs#L16-L20